syntax = "proto3";

package creative_campaign;

// ImageGenerateRequest is sent when images need to be generated
message ImageGenerateRequest {
  string campaign_id = 1;
  string locale = 2;
  string product_id = 3;
  
  // Creative content to base the image on
  string headline = 4;
  string description = 5;
  repeated string visual_elements = 6;
  
  // Metadata
  string correlation_id = 10;
  string timestamp = 11;
}

// ImageGenerated is published when an image has been generated
message ImageGenerated {
  string campaign_id = 1;
  string locale = 2;
  string product_id = 3;
  
  // Generated image details
  string image_url = 4;  // DALL-E generated image URL
  string s3_uri = 5;     // S3 URI where image is stored
  string prompt_used = 6; // The prompt sent to DALL-E
  
  // Status
  string status = 7;  // "generated" or "failed"
  string error_message = 8;  // Only set if status is "failed"
  
  // Metadata
  string correlation_id = 10;
  string generated_at = 11;
}
